ORMLite implementerat + databas

jkim931 opened the pull request:
Se till att ni har tankat hem de bibliotek som behövs (fråga jocke), sedan måste ni köra filen "DatabaseConfigUtil.java" för att autogenerera en fil som behövs (hamnar på rätt ställe automatiskt).

jkim931 committed:
Håller på o skapar en databashanterare

jkim931 committed:
Merge branch 'master' of https://github.com/Rovanion/AinAppen

jkim931 committed:
Merge branch 'master' of https://github.com/Rovanion/AinAppen

jkim931 committed:
Databas implementerad, mer i comment

Databasen implementerad, ORMLite-relevant info lades till i
våra dataobjekt för att ORMLite skall kunna fungera. AddCaseFragment
har fått bättre variabelnamn på knapparna/fälten samt en listener
som lägger in case i databasen.

jkim931 committed:
Merge branch 'master' of https://github.com/Rovanion/AinAppen

Conflicts:
	res/layout/fragment_case.xml

jkim931 committed:
Merge branch 'master' of https://github.com/henninghall/AinAppen

jkim931 committed:
databas + scroller implementerad

mergad med den kod som henning har, funkar inte på SSG2 men på SSG4
och i emulator...

jkim931 committed:
Fixade en scroller på case-GUIt

henninghall commented on src/gov/polisen/ainappen/AddCaseFragment.java:
```diff
@@ -67,5 +85,65 @@ private void setupStatusSpinner(View view) {
+	 * TODO: Integrity check input-fields before committing to database
+	 * (i.e crime_class must not be null)
+	 */
+	public void addCaseToDB(){
```
Flytta denna funktion till nån typ av database helper class istället. I AddCaseFragment borde endast en instans skapas av DBHelper och köra dennes funktioner.  Typ så här: 

Case createdCase = new Case(......)
DBHelper db = new DBHelper();
db.addCase(createdCase);

henninghall commented:
Ta bort oanvända importer så blir perfektionisten inom mig glad som slipper se massa warnings :)

henninghall commented:
Fungerar finfint att köra!! Databasen funkar! 

Kan du lägga upp en guide för hur man gör med Java-filen ska som köras för att skapa textfil eller hur det nu var? Så slipper du springa runt till alla och hjälpa folk med det :)

samuelnilsson commented:
Databasen fungerar. Fick en genomgång hur det går till och det verkar smart. Jocke skulle dock strukturera om koden lite så frågan är om han inte ska göra en till pull request med de uppdateringarna innan vi accepterar den här.

Rovanion commented:
Han gör då inte en ny pull request utan pushar bara upp till sin master
branch varpå de inkluderas i den här pull requesten.

Den 26 mars 2014 13:14 skrev samni698 notifications@github.com:

> Databasen fungerar. Fick en genomgång hur det går till och det verkar
> smart. Jocke skulle dock strukturera om koden lite så frågan är om han inte
> ska göra en till pull request med de uppdateringarna innan vi accepterar
> den här.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/Rovanion/AinAppen/pull/13#issuecomment-38676321
> .

Rovanion commented on src/gov/polisen/ainappen/Case.java:
```diff
@@ -1,29 +1,49 @@
-	private Date date; 
-	private String status; 
+	@DatabaseField
+	private String date; 
```
Detta fält borde vara av typ java.sql.Date eller liknande.

Rovanion commented on src/gov/polisen/ainappen/Case.java:
```diff
@@ -1,29 +1,49 @@
+	@DatabaseField
 	private String crimeClassification; 
+	@DatabaseField
 	private String location; 
```
Kolla på hur de med de andra grupperna gemensamma tabellerna ser ut. Location har ett visst format där som vi bör följa även i våra egna databaser för att minska antalet typkonverteringar vi gör.

Du ska ha fått ett epost om det.

Rovanion commented on src/gov/polisen/ainappen/Case.java:
```diff
@@ -1,29 +1,49 @@
+	@DatabaseField
 	private String location; 
+	@DatabaseField
 	private String commander; 
```
Det här bör då enligt konversationen vi just hade vara en int. En int som har användar-di:t på befälet.

jkim931 committed:
Fixed a "LocalDBHelper" + CaseGUIthings

A "LocalDBHelper" has been made for easier access to common add-functions
used by on-click listeners. Fixed unused imports, as well as updated
GUI-functions with addCaseFragment. The CaseListFragment (the fragment
showing all cases found in our database in a nice list) now correctly
shows all the correct Cases.

Also fixed an issue where, after creating a new Case you'd return to the
"add_case_fragment" after pressing return in the case-view. The user now
correctly returns to the case-list instead.

jkim931 committed:
Previous issues from comments fixed

Fixade mycket av det som stod i kommentarerna, när man skapar ärende kan
man nu välja datum ur en calendar view som autodefaultar till dagens
datum. Datatypen för date i 'Case' är ändrad till java.util.Date, en
mindre "bugg" i CaseFragment där fel meddelande visades under fel rubrik
fixades dessutom. Utöver detta finns nu en "LocalDBHelper" som i princip
är ett objekt som "bridgar" gapet mellan db-connection och önskad effekt i
GUI. Detta för att slippa ha flertalet 'komplexa' rader kod i GUI-listeners osv.

Rovanion commented:
Du måste pulla rovanion master och lösa konflikterna som uppstår. Sedan är det nåra inline comments som står kvar.

jkim931 committed:
Merge branch 'master' of https://github.com/Rovanion/AinAppen

Conflicts:
	src/gov/polisen/ainappen/CaseListAdapter.java

jkim931 committed:
Mergat med Rovanion Master! Nemas problemas...

Funkar o köra på mobilen dessutom
